estuary/utils/general.py

```diff
@@ -182,43 +182,53 @@ def create_story_query(item, node_id, reverse=False, limit=False):
     return query
 
 
-def get_correlated_nodes_count(results):
+def get_correlated_nodes_count(results, reverse=False):
     """
     Iterate through the results and yield correlated nodes.
 
     :param list results: contains inflated results from Neo4j
+    :kwarg bool reverse: determines if we are traversing the story/path backwards
     :return: yield the results count (int) received from Neo4j
     :rtype: generator
     """
-    if len(results) < 2:
+    len_story = len(results)
+    if len_story < 2:
         raise RuntimeError('This function can\'t be called with one or zero elements')
-    for index in range(len(results)):
-        # If it's the last node in the story, the siblings for it are determined
-        # depending on the correlated node prior to it in the story
-        if index == len(results) - 1:
-            yield get_correlated_nodes(results[index].__label__, results[index - 1],
-                                       last=True, count=True)
-        else:
-            yield get_correlated_nodes(results[index].__label__, results[index + 1], count=True)
 
+    correlated_nodes = []
+    if not reverse:
+        for index in range(len_story - 1):
+            correlated_nodes.append(get_correlated_nodes(
+                results[index].__label__, results[index + 1], count=True))
+    else:
+        for index in range(len_story - 1, 0, -1):
```

Comment:
Add a comment saying you are iterating over the results backwards for convenience but it'll be reversed to be in the correct order later on.